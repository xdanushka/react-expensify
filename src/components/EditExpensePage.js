import React from 'react';
import { connect} from 'react-redux';
import ExpenseForm from './ExpenseForm';
import  {editExpense, removeExpense} from '../actions/expenses'

export class EditExpensePage extends React.Component{
    onClick = () => {
        this.props.onRemove(this.props.expense.id);
        this.props.history.push('/');
    }

    onSubmit = (expense) => {
        this.props.onEdit(this.props.expense.id, expense);
        this.props.history.push('/');
    }

    render() {
        return (
            <div>
                <ExpenseForm 
                    expense={this.props.expense}
                    onSubmit={this.onSubmit}/>
                <button onClick={this.onClick}>
                    Remove
                </button>
            </div>
        )
    }
}

const mapStateToProps = (state, props) => {
    return {
        expense: state.expenses.find((expense)=>expense.id===props.match.params.id) 
    }
}

const mapDispatchToProps = (dispatch) => {
    return {
        onRemove: (id) => dispatch(removeExpense({id})),
        onEdit: (id, expense) =>dispatch(editExpense(id,expense))
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(EditExpensePage)