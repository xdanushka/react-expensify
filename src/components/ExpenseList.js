import React from 'react';
import {connect}  from 'react-redux';
import getVisibleExpenses from '../selectors/expenses'
import ExpenseListItem from './ExpenseListItem';

export const ExpenseList = (props) => (
    <div>
        {
            props.expenses.length === 0 ? (
                <p>No Expenses</p>
            ) : (
                props.expenses.map((ex)=>{
                    return (<ExpenseListItem key={ex.id} expense={ex} {...ex} />)
                })
            )
        }
    </div>
);

const mapStateToProps = (state) =>{
    return {
        expenses: getVisibleExpenses(state.expenses,state.filters)
    }
}

export default connect(mapStateToProps)(ExpenseList);