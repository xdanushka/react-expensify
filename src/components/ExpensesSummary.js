import React from 'react';
import {connect}  from 'react-redux';
import numaral from 'numeral';
import getTotalExpenses from '../selectors/expenses-total'
import getVisibleExpenses from '../selectors/expenses'

export const ExpensesSummary = (props) => {
    const expenseWord = props.expenseCount == 1 ? 'expense' : 'expenses'
    const formattedExpenseTotal = numaral(props.expenseTotal / 100).format('$0,0.0');
    return (
        <div>
            <h1>Viewing {props.expenseCount} {expenseWord} totalling {formattedExpenseTotal}</h1>
        </div>
    )
}

const mapStateToProps = (state) =>{
    const visibeleExpenses = getVisibleExpenses(state.expenses, state.filters);
    return {
        expenseCount: visibeleExpenses.length,
        expenseTotal: getTotalExpenses(visibeleExpenses)
    }  

}

export default connect(mapStateToProps)(ExpensesSummary);