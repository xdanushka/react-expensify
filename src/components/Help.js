import React from 'react';
const HelpPage = () => (
    <div>
        Help component
    </div>
);

export default HelpPage