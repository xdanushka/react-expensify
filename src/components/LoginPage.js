import React from 'react';
import {connect} from 'react-redux';
import {startLogin} from '../actions/auth';

export const LoginPage = ({startLogin}) => {
    return (
        <div>
            <p> this is login page</p>
            <button onClick={startLogin}>Login</button>
        </div>
        
    )
}

const mapDispatchToProps = (dispatch) => {
    return {
        startLogin: () =>dispatch(startLogin())
    }
}

export default connect(undefined, mapDispatchToProps)(LoginPage);