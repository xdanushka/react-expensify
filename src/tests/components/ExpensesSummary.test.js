import React from 'react';
import {shallow} from 'enzyme'
import { ExpensesSummary } from '../../components/ExpensesSummary';
import expenses from '../fixtures/expenses'

test('should render AddExpensePage with multiple expense', () => {
    const wrapper = shallow(<ExpensesSummary expenseTotal={1000} expenseCount={2}/>)
    expect(wrapper).toMatchSnapshot();
})

test('should render AddExpensePage with one expense', () => {
    const wrapper = shallow(<ExpensesSummary expenseTotal={100} expenseCount={1}/>)
    expect(wrapper).toMatchSnapshot();
})

