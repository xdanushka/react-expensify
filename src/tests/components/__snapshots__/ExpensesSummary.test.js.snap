// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render AddExpensePage with multiple expense 1`] = `
<div>
  <h1>
    Viewing 
    2
     
    expenses
     totalling 
    $10.0
  </h1>
</div>
`;

exports[`should render AddExpensePage with one expense 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     totalling 
    $1.0
  </h1>
</div>
`;
